iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -289,6 +289,12 @@ Interfaces ServiceDescription::getSourceInterface() const noexcept
     return m_interfaceSource;
 }
 
+bool ServiceDescription::isValid() const noexcept
+{
+    return !(m_serviceString == iox::capro::InvalidIDString || m_serviceID == iox::capro::AnyService
+             || m_instanceString == iox::capro::InvalidIDString || m_instanceID == iox::capro::AnyInstance);
```

Comment:
For default ServiceDescription, which is an invalid case, m_serviceID and m_instanceId (represented as u_int ) are initialized with InvalidID which is zero, on other hand for valid ServiceDescription also... say {"service1", "instance1"}  m_serviceID and m_instanceId are also deduced to zero, hence need to check for m_serviceString and m_instanceString against iox::capro::InvalidIDString instead of checking against InvalidID with m_serviceID and m_instanceId.

Reply:
You need to check if the Servicedescription is full blown or not. If the reduced one is constructed then a bool member `m_hasServiceOnlyDescription` is set and can be checked with `hasServiceOnlyDescription()`.

Also when you construct a ServiceDescription{"service1", "instance1"} then the members m_serviceID and m_instanceId are set. 
The ctor of ServiceDescription(IDString, IDString) is redirecting to  ServiceDescription(IDString, IDString, InvalidString). And there is also a conversion from string to uint_16t happening. Therefore you only need to check the uint16_t members. Take a look in the constructors in service_description.cpp

Reply:
Conversion is happening if stringIsNumber, in case of {"service1", "instance1"}, stringIsNumber() returns false, and m_serviceID and m_instanceId are set to InvalidID.

Also i dint get why i need to check for Servicedescription is full blown or not, I guess we are not checking for event, do u want to consider for future case where event may consider ??

Reply:
Yes you are right, i somehow didn't want to accept that the term "service1" can't be represented as int, shame on me. :-) Then it is the safest way to compare the strings. 
What if you are internally using full blown service descriptions like the ara::com API? This could cause trouble for the case that someone constructs a full blown servicedescription with `m_hasServiceOnlyDescription = false` and someone sets the `m_eventID` to zero. It is maybe a special case but can rarely happen.

Reply:
okay, let me share my understanding.
If full blown service description is constructed and `m_eventID` sets to zero then it should be considered as invalid case and should be ignored, on other hand for description like {"service1", "instance1"}, where `m_hasServiceOnlyDescription` will be true and m_`eventID` will be zero should be considered as valid case. This is what do you mean? @dkroenke

Reply:
@nihalchari Yes correct